src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -233,6 +236,26 @@ EmberAfStatus OnOffServer::setOnOffValue(chip::EndpointId endpoint, chip::Comman
                 status = ModeSelect::Attributes::CurrentMode::Set(endpoint, onMode.Value());
             }
         }
+#endif
+#ifdef EMBER_AF_PLUGIN_MODE_BASE
+        // If OnMode is not a null value, then change the current mode to it.
+        for (const auto & element : ModeBase::Instance::ModeBaseAliasesInstanceMap)
```

Comment:
This has a global instance. Suggest replacing `ModeBase::Instance` with `ModeBase::GetInstance()` and never directly refer to the instance. This will allow moving the instance elsewhere in the future.

Also, would be better to have a const-ref-getter for ModeBaseAliasesInstanceMap instead of direct global access.

Both can be done  in follow-up (please add a TODO with issue link).

Reply:
I had actually done this and removed it for some reason. I'll add it back in.

Reply:
I moved the map to the ModeBase namespace in [this commit](https://github.com/project-chip/connectedhomeip/pull/27504/commits/fc2081b7876c15b14542ca73ccc8aa88bf0c52ad) and logged a [feature issue](https://github.com/project-chip/connectedhomeip/issues/27508) to provide better hooks for the OnOff cluster.